### PR TITLE
[skin] Reworked DialogPlayerProcess to add more info

### PR DESCRIPTION
**Pull request by CastagnaIT**
## Description

This will add more infos to the current Player Process Info window dialog,
now there are four sections:
- PVR -> to see PVR info (unchanged)
- Media -> to see metadata infos
- Player -> (the previous `Player Process Info`) to see infos processed by the kodi player
- System -> to check system status

the name of the xml (DialogPlayerProcessInfo.xml) should be renamed (in future on other PR) appropriately to better reflect its functionality, since more information is now being provided not only for `Player Process Info`

In estuary skin, i have also made a small fix to keep the information visible on the screen even when moving the mouse pointer.

## Motivation and context

Initially for the need to see in real time when the video bitrate change while watching a video with InputStreamAdaptive,
then has been discussed (#20593) to reorganise the information in the most appropriate way

This superseed PR's:
#20593
#11693

## How has this been tested?

by using InputStreamAdaptive, see screenshots

## What is the effect on users?

More useful information

## Screenshots (if appropriate):
Estouchy
immagine
Estuary
immagine
immagine
immagine

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by enen92** [comment]
Adding @ksooo and @ronie as this affects PVR and skinning, your input is greatly appreciated.
Thanks for the improvement @CastagnaIT and @jjd-uk

**Comment by jjd-uk** [comment]
Just a quick acknowledgement that I've seen the request for review. I will take a look at this over the weekend.

**Review by ronie — CHANGES_REQUESTED** [review]

File: addons/skin.estouchy/language/resource.language.en_gb/strings.po [hunk 1]
```diff
@@ -78,22 +78,27 @@ msgctxt "#31016"
 msgid "Albums"
 msgstr ""
 
+#: /xml/DialogPlayerProcessInfo.xml
```
Comment (ksooo) [inline addons/skin.estouchy/language/resource.language.en_gb/strings.po:81]: Please add verbal descriptions for the labels to make the life of our translaters easier.

`#. Label for ...`

**Review by ksooo — CHANGES_REQUESTED** [review]
Please add verbal descriptions for all the labels touched by this PR to make the life of our translaters easier.

File: addons/skin.estouchy/language/resource.language.en_gb/strings.po [hunk 2]
```diff
@@ -78,22 +78,27 @@ msgctxt "#31016"
...
 msgctxt "#31017"
-msgid "PVR info"
+msgid "PVR"
 msgstr ""
 
+#: /xml/DialogPlayerProcessInfo.xml
 msgctxt "#31018"
-msgid "Player process info"
+msgid "Player"
```
Comment (jjd-uk) [inline addons/skin.estouchy/language/resource.language.en_gb/strings.po:88]: You got 2 labels of string "Player" now as you also added 31611

File: addons/skin.estouchy/language/resource.language.en_gb/strings.po [hunk 3]
```diff
@@ -386,7 +391,74 @@ msgctxt "#31565"
...
+msgstr ""
+
+#: /xml/DialogPlayerProcessInfo.xml
+msgctxt "#31610"
+msgid "Media"
+msgstr ""
+
+#: /xml/DialogPlayerProcessInfo.xml
+msgctxt "#31611"
+msgid "Player"
```
Comment (jjd-uk) [inline addons/skin.estouchy/language/resource.language.en_gb/strings.po:453]: Duplicate of 31138

File: addons/skin.estuary/language/resource.language.en_gb/strings.po [hunk 4]
```diff
@@ -641,12 +642,12 @@ msgstr ""
...
 #: /xml/DialogPlayerProcessInfo.xml
 msgctxt "#31137"
-msgid "PVR info"
+msgid "PVR"
 msgstr ""
 
 #: /xml/DialogPlayerProcessInfo.xml
 msgctxt "#31138"
-msgid "Player process info"
+msgid "Player"
```
Comment (jjd-uk) [inline addons/skin.estuary/language/resource.language.en_gb/strings.po:650]: Same as before duplicate string "Player" also been added as 31611

File: addons/skin.estuary/language/resource.language.en_gb/strings.po [hunk 5]
```diff
@@ -807,3 +808,70 @@ msgstr ""
...
+msgstr ""
+
+#: /xml/DialogPlayerProcessInfo.xml
+msgctxt "#31610"
+msgid "Media"
+msgstr ""
+
+#: /xml/DialogPlayerProcessInfo.xml
+msgctxt "#31611"
+msgid "Player"
```
Comment (jjd-uk) [inline addons/skin.estuary/language/resource.language.en_gb/strings.po:871]: Duplicate of 31138

File: addons/skin.estuary/xml/DialogPlayerProcessInfo.xml [hunk 6]
```diff
@@ -212,12 +316,28 @@
...
 					<textoffsetx>40</textoffsetx>
-					<label>$LOCALIZE[31138]</label>
+					<label>$LOCALIZE[31610]</label>
+					<onfocus>SetProperty(PlayerInfoDialogFocus,5552,Home)</onfocus>
+				</control>
+				<control type="button" id="5553">
+					<width>auto</width>
+					<height>100</height>
+					<textoffsetx>40</textoffsetx>
+					<label>$LOCALIZE[31611]</label>
```
Comment (jjd-uk) [inline addons/skin.estuary/xml/DialogPlayerProcessInfo.xml:332]: Using the new duplicate 31611 when you changed the previous 31138

**Comment by jjd-uk** [comment]
Skin changes look good to me.

One thing I've discover though when testing on a current Windows x64 Nightly is that $INFO[VideoPlayer.VideoBitrate] doesn't return anything. My previous tests on my work PC were with v19 so it seems something may have changed in v20 so will investigate further. Not directly related to this PR however, but thought I'd mention it.

image

File: addons/skin.estouchy/xml/DialogPlayerProcessInfo.xml [hunk 7]
```diff
@@ -204,20 +310,34 @@
...
+					<include>ButtonInfoDialogsCommonValues</include>
+					<label>$LOCALIZE[31018]</label>
+					<onclick>SetProperty(PlayerInfoDialogFocus,5553,Home)</onclick>
+				</control>
+				<control type="button" id="5554">
+					<width>auto</width>
+					<height>100</height>
+					<include>ButtonInfoDialogsCommonValues</include>
+					<label>$LOCALIZE[31612]</label>
+					<onclick>SetProperty(PlayerInfoDialogFocus,5554,Home)</onclick>
```
[the remark this answers is not in the document]
Comment (CastagnaIT) [reply addons/skin.estouchy/xml/DialogPlayerProcessInfo.xml:340]: i had forgotten thanks

File: addons/skin.estouchy/language/resource.language.en_gb/strings.po (hunk 1 above)
Comment (CastagnaIT) [reply addons/skin.estouchy/language/resource.language.en_gb/strings.po:81]: done

File: addons/skin.estouchy/language/resource.language.en_gb/strings.po (hunk 2 above)
Comment (CastagnaIT) [reply addons/skin.estouchy/language/resource.language.en_gb/strings.po:90]: oversight thanks fixed

**Review by enen92 — COMMENTED** [review]
Looks good to me. Thanks.
Some other team member must approve before merging though

**Review by enen92 — APPROVED** [review]

**Review by jjd-uk — APPROVED** [review]
All changes now look good to me

**Comment by jjd-uk** [comment]
Btw I was thinking whether we should make this revised dialog more easily accessible by adding an option to the Settings menu (perhaps that should be renamed to something like **Options**)

image

Named it **Technical info** as I couldn't think of anything better to cover the different type of info it now presents.

If wanted I'll create a PR for it.

**Comment by enen92** [comment]
@jjd-uk can we somehow restrict the items shown by settings level (e.g. only show if level is expert) or is this just a select dialog with items added from the core?

**Comment by jjd-uk** [comment]
> @jjd-uk can we somehow restrict the items shown by settings level (e.g. only show if level is expert) or is this just a select dialog with items added from the core?

Not as far as I'm aware as there's no Infolabel/Infobool to return the selected selected Settings Level. This Settings list is a custom dialog https://github.com/xbmc/xbmc/blob/master/addons/skin.estuary/xml/Custom_1101_SettingsList.xml that uses a grouplist of buttons.

**Comment by enen92** [comment]
Hum, then I think that likely should be added first to avoid cluttering dialogs with "expert" info for simple use cases and basic installations. I doubt the majority of users are interested in checking computational resource usage or player codecs/bitrates

File: addons/skin.estouchy/language/resource.language.en_gb/strings.po (hunk 1 above)
Comment (ksooo) [reply addons/skin.estouchy/language/resource.language.en_gb/strings.po:81]: > done

I don't see any descriptions for the new/changed labels in the now merged PR!

**Pull request merged**